control_panel_api/authentication.py

```diff
@@ -1,37 +1,70 @@
+import logging
+
+import requests
 from django.conf import settings
-import jwt
-from jwt.exceptions import InvalidTokenError
+from jose import jwt
+from jose.exceptions import JWTError
+from requests.exceptions import RequestException
 from rest_framework.authentication import (
     BaseAuthentication,
-    get_authorization_header
+    get_authorization_header,
 )
 from rest_framework.exceptions import AuthenticationFailed
 
 from control_panel_api.models import User
 
+logger = logging.getLogger(__name__)
 
-class Auth0JWTAuthentication(BaseAuthentication):
 
-    def authenticate(self, request):
+def get_jwks():
+    response = requests.get(
```

Comment:
I suspect it's not necessarily safe to hit this URL directly, as the JWKS URL isn't part of the OIDC spec. The configuration URL is though, and the response from that contains the JWKS URL:

https://dev-analytics-moj.eu.auth0.com/.well-known/openid-configuration -

`"jwks_uri":"https://dev-analytics-moj.eu.auth0.com/.well-known/jwks.json"`

So it's better to hit the `openid-configuration` URL first, and get the JWKS URL from the response.

Reply:
Actually I think I'm wrong about this - looking at Auth0's own examples, they hit the JWKS endpoint directly - https://github.com/auth0/node-jwks-rsa#usage - and it's also probably safe to assume that if Auth0 change the JWKS endpoint URL they'd give us warning well in advance.

Reply:
Ok, cool.

Reply:
@andyhd I thought that, but there doesn't seem to be any expiry times on the JWKS? https://auth0.com/docs/jwks